lib/lightning/runs/run.ex

```diff
@@ -95,10 +96,14 @@ defmodule Lightning.Run do
     |> put_assoc(:starting_trigger, trigger)
     |> put_assoc(:dataclip, attrs[:dataclip])
     |> put_assoc(:snapshot, attrs[:snapshot])
-    |> add_options(attrs.dataclip.project_id)
     |> validate_required_assoc(:dataclip)
     |> validate_required_assoc(:snapshot)
     |> validate_required_assoc(:starting_trigger)
+    |> then(fn changeset ->
+      snapshot = get_assoc(changeset, :snapshot, :struct)
+      dataclip = get_assoc(changeset, :dataclip, :struct)
+      add_options(changeset, snapshot.workflow_id, dataclip.project_id)
+    end)
```

Comment:
I don't think we should be calling the UsageLimiter (via `add_options` from inside a the schema's changeset; we want to keep these functions side effect free.

It's the responsibility of the calling module to create a run with the correct options, and the validity of the params is decided by this (the Run) module.

Reply:
Riight. Let me work on moving it out